src/main/java/mezz/jei/gui/ghost/GhostIngredientDrag.java

```diff
@@ -75,22 +78,35 @@ public void drawItem(Minecraft minecraft, PoseStack poseStack, int mouseX, int m
 				float percentOfDim = Math.min(1, distanceSq / (float) screenDim);
 				lineWidth = 1 + ((1 - (percentOfDim)) * 3);
 			}
-			GL11.glDisable(GL11.GL_TEXTURE_2D);
+
+			RenderSystem.disableTexture();
 			RenderSystem.disableDepthTest();
+			RenderSystem.depthMask(false);
+
+			var oldShader = RenderSystem.getShader();
+			RenderSystem.setShader(GameRenderer::getPositionColorShader);
+
 			GL11.glLineWidth(lineWidth);
 			GL11.glEnable(GL11.GL_LINE_SMOOTH);
 			GL11.glHint(GL11.GL_LINE_SMOOTH_HINT, GL11.GL_NICEST);
-			GL11.glBegin(GL11.GL_LINES);
+
+			var tesselator = RenderSystem.renderThreadTesselator();
+			var builder = tesselator.getBuilder();
+			builder.begin(VertexFormat.Mode.DEBUG_LINES, DefaultVertexFormat.POSITION_COLOR);
 			float red = (targetColor >> 24 & 255) / 255.0F;
 			float green = (targetColor >> 16 & 255) / 255.0F;
 			float blue = (targetColor >> 8 & 255) / 255.0F;
 			float alpha = (targetColor & 255) / 255.0F;
-			RenderSystem.setShaderColor(red, green, blue, alpha);
-			GL11.glVertex3f(mouseX, mouseY, 150);
-			GL11.glVertex3f(originX, originY, 150);
-			GL11.glEnd();
+			builder.vertex(originX, originY, 150).color(red, green, blue, alpha).endVertex();
+			builder.vertex(mouseX, mouseY, 150).color(red, green, blue, alpha).endVertex();
```

Comment:
Is there a reason this is changing it from drawing from mouse to origin, to drawing from origin to mouse? I doubt it really matters just would be probably worth mentioning in this PR why this change is being made.

Reply:
No reason, just by accident. I also don't think it matters for lines since they should not be culled in any case (i know of, at least).

Reply:
Reverted this back to its original order as part of removing line width.